### DISCUSSION TITLE
Can't reach database server at

### DISCUSSION THREAD

**Discussion by maxganiev** [prisma/prisma#15005] [Q&A] [answered]
hi everyone!
I noticed that this issue is kinda oftenly faced, thus I am really surprised there are still so many of them unanswered.
Basically, I got exactly same error when attempting to deploy my fullstack app to nginx server. And while on local it works as expected, on a server I am keep getting same message.
I am 100% sure the db lives on host 176.57.220.55 and port 443, we've got several websites connected to it atm, ex. see screenshot below - this is image received from the server the db lives at.
![image](https://user-images.githubusercontent.com/63904631/186689349-913bfa3e-8dd8-4a71-9a31-0354f2567cc4.png)

and this is website url https://newel-industry.com/client/dist/

got no idea what shall be done, spent a whole day in attemps to figuring out. need help
![image](https://user-images.githubusercontent.com/63904631/186689747-7edaa955-11d4-4a13-9ef2-16ceb19b3539.png)

**Comment by austincrim** [comment]
Hey @maxganiev 👋,

Have you double-checked your username:password combination? Do you need to U[RL-encode any special characters](https://www.prisma.io/docs/reference/database-reference/connection-urls#special-characters)?

**Reply by maxganiev** [reply]
> Hey @maxganiev 👋,
> 
> 
> 
> Have you double-checked your username:password combination? Do you need to U[RL-encode any special characters](https://www.prisma.io/docs/reference/database-reference/connection-urls#special-characters)?

Hello @austincrim 
Yes I double and triple checked everything. In fact, as I mentioned above, other sites are connected to this very db atm, however in related db connection config files all details were provided but port (bzw, host is pointed as localhost). Those sites send requests via laravel though and one via node js mysql clean syntax, so the only problem emerged with prisma.
Bit frustrated already tbh

My credentials don't contain any special characters

**Reply by maxganiev** [reply]
@austincrim in fact i have managed to get away from dead point... in fact the problem was in mysql itself, i have fixed it and now there is another issue 🤯 
first request is sent OK, however 2nd one returns auth error for some reason...
![image](https://user-images.githubusercontent.com/63904631/186840181-e754ca13-a835-4fbe-afb6-9efe7c28553f.png)

any thoughts on why this can happen?

**Comment by maxganiev** [comment, marked as answer]
So everything is fixed, thanks God. if you are interested, the 1st problem, where db was failed to be find on host and port, occured due to error config in mysql (it basically was closed for external connection); still wondering how php or laravel managed to bypass it, but anyway.
and 2nd problem was with my .env. I basically created 2 different vars for credentials inside of .env file, and then told node to pick up the proper one depending on mode (dev or prod) and consequently to point prisma to proper db. Despite it worked perfectly while on localhost, on a server it managed to successfully resolve only with a first single request; all consequent requests returned auth error.
So to conclude:
1. mysql has to be opened for connections (can be achieved via setting skip_networking=0 in /etc/mysql/mariadb.conf.d/50-server.cnf file (i use ubuntu); you can then check if it is opened by sql command SHOW VARIABLES LIKE "PORT" - it should return anything but 0;
2. create 2 separate .env files for prod and dev and to point prisma to each of those, respectively.

**Comment by zhermin** [comment]
If you've tried all of the solutions mentioned here and elsewhere, and you're spinning up a database and NextJS (with or without Prisma, and with a "server-side" framework) app with docker-compose, then maybe this obscure, headache-inducing gotcha might be the cause, as it was for me. 

# SOLUTION: DISABLE SSR

For me, everything worked fine on "dev" — the database url works and can connect fine. But on "prod", Prisma kept failing to connect to the database during NextJS's build. Healthchecks unfortunately don't work as that only helps to delay a service from starting, not building. 

Turns out, I was using NextJS's `getStaticPaths`, which tries to pre-render pages during build time, in which it tries pulling data from the database, which will never be ready in time as all services are built in parallel. This also only gets triggered in prod mode for optimization purposes. 

Disabling other SSR or server-side features could be the key here. For `getStaticPaths`, a flag can be used to skip the pre-rendering. For other SSR features, you can find other methods elsewhere. 

```typescript
  if (process.env.SKIP_BUILD_STATIC_GENERATION) {
    return {
      paths: [],
      fallback: 'blocking',
    }
  }
```

# TLDR
Disable SSR if using a framework with server-side features that depend on a running database during build-time, if the database is also being built at the same time; tweaking docker-compose or the database will likely lead to a dead end.

Hope this helps someone out!!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
